### PR TITLE
Conversion tests for toMsg()

### PR DESCRIPTION
**Pull request by gleichdick**
Vectors can be converted to a Vector message or a Point Message, some libraries behave differently. This unit test will make sure that the default return value for `toMsg()` will stay the same.

This was separated from #368.

### CODE REVIEW

File: test_tf2/CMakeLists.txt [hunk 1]
```diff
@@ -24,18 +24,19 @@ find_package(geometry_msgs REQUIRED)
...
 
-# TODO (ahcorde): activate when tf2_bullet is merged
-# find_package(PkgConfig REQUIRED)
-# pkg_check_modules(bullet REQUIRED bullet)
-# include_directories(include ${bullet_INCLUDE_DIRS})
+if(WIN32)
+  set(BULLET_ROOT $ENV{ChocolateyInstall}/lib/bullet)
+endif()
+find_package(Bullet REQUIRED)
+include_directories(include ${BULLET_INCLUDE_DIRS})
```
Comment (clalancette) [inline test_tf2/CMakeLists.txt]: Do we actually need any of this?  We aren't using bullet directly in this package, and hence all of this should be inherited from `tf2_bullet`.  If it is not, then that seems like a bug in the `tf2_bullet` package.

File: test_tf2/test/test_convert.cpp [hunk 2]
```diff
@@ -96,7 +97,91 @@ TEST(tf2Convert, kdlBulletROSConversions)
   EXPECT_NEAR(b1.z(), b4.z(), epsilon);
 }
 
-int main(int argc, char** argv)
+TEST(tf2Convert, ConvertTf2Quaternion)
+{
+  const tf2::Quaternion tq(1, 2, 3, 4);
+  Eigen::Quaterniond eq;
```
Comment (clalancette) [inline test_tf2/test/test_convert.cpp:110]: Add needed includes for these at the top:

```
#include <tf2/LinearMath/Quaternion.h>
#include <Eigen/Eigen>
```

Also, I think we need to declare a dependency on `eigen` in the package.xml and the CMakeLists.txt.

File: test_tf2/test/test_convert.cpp [hunk 3]
```diff
@@ -96,7 +97,91 @@ TEST(tf2Convert, kdlBulletROSConversions)
...
+  //tf2::convert(tq, eq);
```
Comment (clalancette) [inline test_tf2/test/test_convert.cpp]: Remove this dead code.

File: test_tf2/test/test_convert.cpp [hunk 4]
```diff
@@ -96,7 +97,91 @@ TEST(tf2Convert, kdlBulletROSConversions)
...
+}
+
+TEST(tf2Convert, PointVectorDefaultMessagetype)
+{
+  // Verify the return type of `toMsg()`
+  // as it can return a Vector3 or a Point for certain datatypes
+  {
+    // Bullet
+    const tf2::Stamped<btVector3> b1{btVector3{1.0, 3.0, 4.0}, tf2::TimePoint(), "my_frame"};
+    const geometry_msgs::msg::PointStamped msg = tf2::toMsg(b1);
```
Comment (clalancette) [inline test_tf2/test/test_convert.cpp:127]: Add the include at the top of the file:

```
#include <geometry_msgs/msg/point_stamped.hpp>
```

File: test_tf2/test/test_convert.cpp [hunk 5]
```diff
@@ -96,7 +97,91 @@ TEST(tf2Convert, kdlBulletROSConversions)
...
+
+    EXPECT_EQ(msg.point.x, 1.0);
+    EXPECT_EQ(msg.point.y, 3.0);
+    EXPECT_EQ(msg.point.z, 4.0);
+    EXPECT_EQ(msg.header.frame_id, b1.frame_id_);
+  }
+  {
+    // Eigen
+    const Eigen::Vector3d e1{2.0, 4.0, 5.0};
+    const geometry_msgs::msg::Point msg = tf2::toMsg(e1);
```
Comment (clalancette) [inline test_tf2/test/test_convert.cpp:137]: Add the include at the top of the file.

File: test_tf2/test/test_convert.cpp [hunk 6]
```diff
@@ -96,7 +97,91 @@ TEST(tf2Convert, kdlBulletROSConversions)
...
+
+    EXPECT_EQ(msg.x, 2.0);
+    EXPECT_EQ(msg.y, 4.0);
+    EXPECT_EQ(msg.z, 5.0);
+  }
+  {
+    // tf2
+    const tf2::Vector3 t1{2.0, 4.0, 5.0};
+    const geometry_msgs::msg::Vector3 msg = tf2::toMsg(t1);
```
Comment (clalancette) [inline test_tf2/test/test_convert.cpp:146]: Add the include at the top of the file.

File: tf2_geometry_msgs/include/tf2_geometry_msgs/tf2_geometry_msgs.hpp [hunk 7]
```diff
@@ -60,6 +61,35 @@ KDL::Frame gmTransformToKDL(const geometry_msgs::msg::TransformStamped& t)
...
+/** Vector3 **/
+/*************/
+
+/** \brief Convert a tf2 Vector3 type to its equivalent geometry_msgs representation.
+ * This function is a specialization of the toMsg template defined in tf2/convert.h.
+ * \param in A tf2 Vector3 object.
+ * \return The Vector3 converted to a geometry_msgs message type.
+ */
+inline
+geometry_msgs::msg::Vector3 toMsg(const tf2::Vector3& in)
```
Comment (clalancette) [inline tf2_geometry_msgs/include/tf2_geometry_msgs/tf2_geometry_msgs.hpp]: Add the include for `geometry_msgs::msg::Vector3` at the top of the file.

Also, style nit: there should be a space before and after the reference symbol (`&`).  We aren't enforcing it in this package yet (and I don't think we should in this PR), but we may as well make one less thing we have to change later.

The same comment goes several other times below.

File: tf2_geometry_msgs/include/tf2_geometry_msgs/tf2_geometry_msgs.hpp [hunk 8]
```diff
@@ -124,6 +154,35 @@ void fromMsg(const geometry_msgs::msg::Vector3Stamped& msg, geometry_msgs::msg::
...
+/** Point **/
+/***********/
+
+/** \brief Convert a tf2 Vector3 type to its equivalent geometry_msgs representation.
+ * This function is a specialization of the toMsg template defined in tf2/convert.h.
+ * \param in A tf2 Vector3 object.
+ * \return The Vector3 converted to a geometry_msgs message type.
+ */
+inline
+geometry_msgs::msg::Point& toMsg(const tf2::Vector3& in, geometry_msgs::msg::Point& out)
```
Comment (clalancette) [inline tf2_geometry_msgs/include/tf2_geometry_msgs/tf2_geometry_msgs.hpp]: Add the include for `geometry_msgs::msg::Point` at the top of the file.

**Review by clalancette — CHANGES_REQUESTED** [review]

File: test_tf2/CMakeLists.txt (hunk 1 above)
Comment (gleichdick) [reply test_tf2/CMakeLists.txt]: Yes, there seems to be a bug. #428 opened

**Comment by gleichdick** [comment]
I hope I addressed all of the remarks. There seems to be a blocking issue with bullet dependencies, see #428.

**Review by clalancette — APPROVED** [review]
This looks good.  I'll run CI on it next.

**Comment by clalancette** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14701/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9453/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12366/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14845/)

**Pull request merged**